audit eslint / prettier configs - not catching uninitialized class members

Check and make sure that the eslint/prettier configs are up to date with what we've been using internally.

In particular - I noticed that if I was writing a class that had some private readonly member variables, and those variables were not initialized in the constructor, I didn't get a build/lint error.  In our CDK repos I believe we would catch that at compile time.

@cprice404 This appears to be a typescript thing governed by [`noUnusedLocals`](https://www.typescriptlang.org/tsconfig#noUnusedLocals).

[We have turned it off](https://github.com/search?p=1&q=org%3Amomentohq+noUnusedLocals&type=Code) everywhere but [momentohq/serverless-api-demo/infra/aws/tsconfig.json](https://github.com/momentohq/serverless-api-demo/blob/a07236239876dd781d209ca588c7b76ead233fdd/infra/aws/tsconfig.json).

Turning it on in client-sdk-javascript reveals a couple issues.

```
src/errors/errors.ts:58:20 - error TS6133: '_stack' is declared but its value is never read.

58   private readonly _stack: string | undefined;
                      ~~~~~~

src/simple-cache-client.ts:49:20 - error TS6133: 'logger' is declared but its value is never read.

49   private readonly logger: Logger;
                      ~~~~~~
```

@bruuuuuuuce @eaddingtonwhite Do you know why noUnusedLocals is off?

--- a/src/errors/errors.ts
+++ b/src/errors/errors.ts
@@ -55,7 +55,6 @@ export abstract class SdkError extends Error {
   protected readonly _errorCode: MomentoErrorCode;
   protected readonly _messageWrapper: string;
   private readonly _transportDetails: MomentoErrorTransportDetails;
-  private readonly _stack: string | undefined;
   constructor(
     message: string,
     code = 0,

--- a/src/simple-cache-client.ts
+++ b/src/simple-cache-client.ts
@@ -34,7 +34,7 @@ import {
   CacheSetRemoveElements,
   CacheSetRemoveElement,
 } from '.';
-import {getLogger, initializeMomentoLogging, Logger} from './utils/logging';
+import {initializeMomentoLogging} from './utils/logging';
 import {range} from './utils/collections';
 import {SimpleCacheClientProps} from './simple-cache-client-props';
 
@@ -54,14 +54,12 @@ export class SimpleCacheClient {
   private readonly dataClients: Array<CacheClient>;
   private nextDataClientIndex: number;
   private readonly controlClient: ControlClient;
-  private readonly logger: Logger;
 
   /**
    * Creates an instance of SimpleCacheClient.
    */
   constructor(props: SimpleCacheClientProps) {
     initializeMomentoLogging(props.configuration.getLoggerOptions());
-    this.logger = getLogger(this);
     this.configuration = props.configuration;
     this.credentialProvider = props.credentialProvider;
 

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -9,7 +9,7 @@
     "strictNullChecks": true,
     "noImplicitThis": true,
     "alwaysStrict": true,
-    "noUnusedLocals": false,
+    "noUnusedLocals": true,
     "noUnusedParameters": false,
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": false,

